### PR TITLE
Rhmap 17173 render studio

### PR DESCRIPTION
**Pull request by witmicko**
@jhellar 
https://issues.jboss.org/browse/RHMAP-13689

Updated function responsible for updating submissions.

### CODE REVIEW

File: lib/impl/submitFormDataFunctions/processSubmissionUpdates.js [hunk 1]
```diff
@@ -2,39 +2,59 @@ var _ = require('underscore');
...
+          var sameSection = existingField.sectionIndex === updatedField.sectionIndex;
+          missingSection = existingField.sectionIndex === 0 && !updatedField.sectionIndex;
+
+          //exact match for field id and its section index
+          return sameId && sameSection;
+        });
+
+        if (existingFieldData) {
+          existingFieldData.fieldValues = updatedField.fieldValues ;
+          save(formSubmission, existingFieldData);
```
Comment (fh-qube) [inline lib/impl/submitFormDataFunctions/processSubmissionUpdates.js]: MAJOR Move the declaration of "save" before this usage. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AVariableDeclarationAfterUsage)

**Comment by fh-qube** [comment]
SonarQube analysis reported 1 issue:
* MAJOR 1 major

Watch the comments in this conversation to review them.

**Comment by jhellar** [comment]
Verified functionality in local VM.

**Comment by jhellar** [comment]
Verified PDF render in local VM.

**Comment by fh-qube** [comment]
SonarQube analysis reported 2 issues:
* MINOR 1 minor
* INFO 1 info

Watch the comments in this conversation to review them.
Note: the following issues could not be reported as comments because they are located on lines that are not displayed in this pull request:
* MINOR [Function has a complexity of 24 which is greater than 20 authorized.](https://github.com/feedhenry/fh-forms/blob/5d16a9b3ddf613535a91dcb251c18241fce50f04/lib/impl/notification.js#L35) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AFunctionComplexity)
* INFO [Move this trailing comment on the previous empty line.](https://github.com/feedhenry/fh-forms/blob/5d16a9b3ddf613535a91dcb251c18241fce50f04/lib/impl/notification.js#L160) [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3ATrailingComment)

File: lib/impl/submissionExportHelper/csvHeaders.js [hunk 2]
```diff
@@ -109,18 +111,36 @@ function generateCSVHeaders(fieldKeys, mergedFieldEntries, fieldHeader) {
...
+  var fieldKeysProcessed = [];
+
   fieldKeys.forEach(function(fieldKey) {
-    // check if its a repeating form (and extra headers required)
+  // check if its a repeating form (and extra headers required)
     var field = mergedFieldEntries[fieldKey];
+    if (field.type === 'sectionBreak' && field.repeating) {
+      var fieldsInThisSection = sectionUtils.getFieldsInSection(field._id, _.values(mergedFieldEntries));
+      for (var i = 0; i < field.fieldOptions.definition.maxRepeat; i++) {
+        _.each(fieldsInThisSection, function(fieldInThisSection) {
```
Comment (fh-qube) [inline lib/impl/submissionExportHelper/csvHeaders.js]: INFO Define this function outside of a loop. [rule](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AFunctionDefinitionInsideLoop)

**Comment by fh-qube** [comment]
SonarQube analysis reported 1 issue:
* INFO 1 info

Watch the comments in this conversation to review them.

**Pull request merged**